pkg/util/aad/aad.go

```diff
@@ -58,13 +62,14 @@ func GetToken(ctx context.Context, log *logrus.Entry, oc *api.OpenShiftCluster,
 		err = token.RefreshWithContext(ctx)
 		if err != nil {
 			isAADSTS700016 := strings.Contains(err.Error(), "AADSTS700016")
+			isTokenRefreshError := autorest.IsTokenRefreshError(err)
 
 			// populate err with a user-facing error that will be visible if
 			// we're not successful.
 			log.Info(err)
 			err = api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidServicePrincipalCredentials, "properties.servicePrincipalProfile", "The provided service principal credentials are invalid.")
 
-			if isAADSTS700016 {
+			if !isTokenRefreshError || isAADSTS700016 {
```

Comment:
I think this is worth a try, but we need some logging.

Reply:
@jim-minter there is `log.Info(err)` above which logs the original error.

Reply:
good point, well made.